src/index.ts

```diff
@@ -385,18 +396,78 @@ export async function cli(args: string[]) {
     namedExports: undefined,
     webDependencies: undefined,
   };
-  const doesWhitelistExist = !!webDependencies;
-  const arrayOfDeps = webDependencies || Object.keys(pkgManifest.dependencies || {});
+  let doesWhitelistExist = !!webDependencies;
+  let arrayOfDeps = webDependencies || Object.keys(pkgManifest.dependencies || {});
   const hasBrowserlistConfig =
     !!pkgManifest.browserslist ||
     !!process.env.BROWSERSLIST ||
     fs.existsSync(path.join(cwd, '.browserslistrc')) ||
     fs.existsSync(path.join(cwd, 'browserslist'));
+  let dependencyTree: string | undefined;
+
+  // auto detection with --entry flag
+  if (typeof entry === 'string' && entry.length) {
```

Comment:
can you pull this whole chunk into a helper function? basically something that would allow us to do:

```
const arrayOfDeps = [
  ...Object.keys(pkgManifest.dependencies || {}), // allow people to explicitly add
  ...detectDependencyImports(entries), // implicitly add via imports
]
```